packages/lesswrong/server/resolvers/arbitalPageData.ts

```diff
@@ -24,69 +31,125 @@ async function getArbitalPageData(pageAlias = "hyperexistential_separation") {
+  if (!rawRoomData) return null
+  let processedData: AnyBecauseTodo;
+  try {
+    processedData = JSON.parse(rawRoomData)
+  } catch(e) {
+    throw new Error(`Received invalid JSON for Arbital hover preview for page "${pageAlias}"`);
+  }
+  if (!processedData?.pages) return null;
+  const page: any = Object.values(processedData.pages).find((page: any) => page?.alias === pageAlias)
+  if (!page) return null
+  const textField = page.summaries?.Summary || page.clickbait
+  const fixedMarkdown = textField.replace(
+    /\[([^\]]+)\]/g,
+    (fullMatch: string, insideBrackets: string) => {
+      if (insideBrackets.startsWith("-")) {
+        // Formatted like [-id]
+        const linkedPageId = insideBrackets.substr(1);
+        const linkedPageAlias = processedData.pages[linkedPageId]?.alias;
+        if (linkedPageAlias) {
+          const linkedPageTitle = processedData.pages[linkedPageId]?.title;
+          const destinationLink = `http://arbital.com/p/${encodeURIComponent(linkedPageAlias)}`;
+          return `[${markdownEscapeLinkText(linkedPageTitle)}](${destinationLink})`;
+        } else {
+          return `[New Page](https://arbital.com/edit/)`
+        }
+      } else {
+        // Formatted like [id title]
+        // Split on the first space
+        const spaceIndex = insideBrackets.indexOf(' ');
+        const linkedPageId = insideBrackets.substr(0, spaceIndex);
+        const linkedText = insideBrackets.substr(spaceIndex+1);
+        const linkedPageAlias = processedData.pages[linkedPageId]?.alias;
+        if (linkedPageAlias) {
+          const destinationLink = `http://arbital.com/p/${encodeURIComponent(linkedPageAlias)}`;
+          return `[${markdownEscapeLinkText(linkedText)}](destinationLink)`;
+        } else {
+          return `[${markdownEscapeLinkText(linkedText)}](https://arbital.com/edit/)`;
+        }
+      }
+    }
+  )
+  let htmlWithLaTeX: string;
+  try {
+    const htmlNoLaTeX = mdi.render(fixedMarkdown)
+    htmlWithLaTeX = await mjPagePromise(htmlNoLaTeX, trimLatexAndAddCSS)
+  } catch(e) {
+    throw new Error(`Error during Arbital hover-preview markdown/LaTeX conversion for "${pageAlias}"`);
+  }
+  return {
+    html: htmlWithLaTeX,
+    title: page.title
+  }
+}
+
+async function getArbitalPageWithCache(pageAlias: string): Promise<{ html: string, title: string }|null> {
+  const now = new Date();
+  const oldestAllowedAge = new Date(now.getTime() - arbitalCacheExpirationMs);
+
+  const cachedResult = await ArbitalCaches.findOne({
+    pageAlias,
+    fetchedAt: { $gt: oldestAllowedAge },
+  });
+  if (cachedResult) {
+    return {
+      title: cachedResult.title,
+      html: cachedResult.sanitizedHtml,
+    };
+  }
+  
+  const result =  await fetchArbitalPageAsHtml(pageAlias);
+  if (!result) return null;
+  void ArbitalCaches.rawInsert({
+    pageAlias,
+    title: result.title,
+    fetchedAt: now,
+    sanitizedHtml: result.html, //TODO: This came out of a markdown conversion; is it safe?
```

Comment:
This is just stashing the same thing we were already returning to users, right?

Reply:
Yep